changelog/580.feature

```diff
@@ -0,0 +1 @@
+make pytest.fail/skip/xfail instances of BaseException
```

Comment:
I think it would be nice to explain the change with some context to end-users, something like:

> Exceptions raised by `pytest.fail`, `pytest.skip` and `pytest.xfail` now subclass `BaseException`, making them harder to be caught unintentionally by normal code.